### ISSUE TITLE
Intersection type

### ISSUE THREAD

**Issue by Mateiadrielrafael** [adisbladis/korora#1]
It would be nice to have a dual to `union` which simply checks that a type satisfies both predicates. 

My internal implementation does something like
```nix
    intersection = l: r:
      k.typedef' "${l.name} ∧ ${r.name}" (v:
        let
          lerr = l v;
          rerr = r v;
        in
        if lerr == null then rerr else
        if rerr == null then lerr else
        "${lerr}\n${rerr}");

```

**Comment by adisbladis** [comment]
This sounds great and strikes me as particularly useful with the `struct` type to implement something like traits.

**Comment by adisbladis** [comment]
Closed by 3a77d4ad661cb4fb923ab7a8a76148b5f99d30bf
